coreax/util.py

```diff
@@ -186,20 +165,21 @@ def solve_qp(kernel_mm: ArrayLike, kernel_matrix_row_sum_mean: ArrayLike) -> Arr
     :param kernel_matrix_row_sum_mean: :math:`m \times 1` array of Gram matrix means
     :return: Optimised solution for the quadratic program
     """
-    # Validate inputs
-    coreax.validation.validate_is_instance(
-        x=kernel_mm, object_name="kernel_mm", expected_type=Array
-    )
-    coreax.validation.validate_is_instance(
-        x=kernel_matrix_row_sum_mean,
-        object_name="kernel_matrix_row_sum_mean",
-        expected_type=Array,
-    )
-
     # Setup optimisation problem - all variable names are consistent with the OSQP
-    # terminology. Begin with the objective parameters
-    q_array = jnp.array(kernel_mm)
-    c = -jnp.array(kernel_matrix_row_sum_mean)
+    # terminology. Begin with the objective parameters.
+    try:
+        q_array = jnp.array(kernel_mm)
```

Comment:
The intent is clearer if `jnp.asarray` is used instead.

Reply:
Agreed and updated